lib/ui.rb

```diff
@@ -2,13 +2,17 @@
 
 class UI
```

Comment:
The `ask_for_field()` method described in your blog would be one way to refactor it, yes.
You made the right call not using it as the sole public method. :+1: 
But why not use it internally?